src-built-in/components/advancedAppCatalog/src/stores/storeActions.js

```diff
@@ -235,10 +235,16 @@ async function addApp(id, cb = Function.prototype) {
 		delete appConfig.window.url;
 	}
 
-	if (typeof app.manifest !== "object") {
-		appConfig.manifest = { ...appConfig };
+	let manifest;
+	try {
+		manifest = JSON.parse(app.manifest);
```

Comment:
If the `app` has already been parsed, `app.manifest` should be an object, which `JSON.parse` will fail against even though the manifest is as defined by the server.

Are we absolutely certain that only a string representation of their manifest will end up here?

Reply:
Te spec defines the prop as a string. If a user's appD server delivers anyting but a string for this prop, they would be going against the spec, and expect a failure. There definitely needs to be an entirely different error if the JSON is not valid, but validation is on my roadmap

Reply:
Talked directly with Joe on this again. I was confused based on the tests I saw yesterday. Turns out when testing he initially had invalid FDC3 structure to mess with the logic here. It was after that he learned it shouldn't be an object but a string by the specification.

All good here.